src/sys/socket/mod.rs

```diff
@@ -778,12 +869,7 @@ impl<'a> ControlMessage<'a> {
         (*cmsg).cmsg_level = self.cmsg_level();
         (*cmsg).cmsg_type = self.cmsg_type();
         (*cmsg).cmsg_len = self.cmsg_len();
-        let data = self.data();
-        ptr::copy_nonoverlapping(
-            data,
-            CMSG_DATA(cmsg),
-            self.len()
-        );
+        self.write_data(CMSG_DATA(cmsg));
```

Comment:
"write" implies I/O.  Can you choose a different name for this method?

Reply:
changed to `copy_to_cmsg_data`. Will this work?